ui/src/views/Projects.vue

```diff
@@ -174,7 +189,80 @@ export default defineComponent({
       searchString: "",
     };
   },
+  computed: {
+    usersOfProjectToEdit: {
+      get(): string[] {
+        return this.projectToEdit.users;
+      },
+      set(users) {
+        this.projectToEdit.users = users;
+      },
+    },
+  },
+  watch: {
+    projectToEdit() {
+      if (this.projectToEditIndex !== -1) {
+        this.usersOfProjectToEdit =
+          this.projects[this.projectToEditIndex].users;
+        this.updateAvailableUsers();
+      }
+    },
+  },
   methods: {
+    updateAvailableUsers() {
+      let availableUsers: StringArray = [];
+      getUsers()
+        .catch((error: string) => {
+          this.errorMessage = processErrorMessages(error, this.$router);
+          return [];
+        })
+        .then((users) => {
+          users.forEach((user) => {
+            if (this.usersOfProjectToEdit.indexOf(user.email) === -1) {
+              availableUsers.push(user.email);
+            }
+          });
+        });
+      this.availableUsers = availableUsers;
+    },
+    updateUsers(event: Event) {
+      const user = event.toString();
+      if (
+        this.projectToEdit.name !== "" &&
+        this.usersOfProjectToEdit.indexOf(user) !== -1
+      ) {
+        this.errorMessage = `User: [${user}] already added to project: [${this.projectToEdit.name}]`;
+      } else if (this.newProject.users.indexOf(user) !== -1) {
+        this.errorMessage = `User: [${user}] already added to new project`;
+      } else if (this.availableUsers.indexOf(user) === -1) {
+        this.userToAdd = user;
+      } else {
+        this.confirmedUser = user;
+        this.proceedUserUpdate();
+      }
+    },
+    proceedUserUpdate() {
+      if (this.confirmedUser === "") {
+        this.confirmedUser = this.userToAdd;
+      }
+      if (this.projectToEditIndex !== -1) {
+        this.projects[this.projectToEditIndex].users.push(this.confirmedUser);
+        this.usersOfProjectToEdit =
+          this.projects[this.projectToEditIndex].users;
+        this.removeFromAvailableUsers(this.confirmedUser);
+      } else {
+        this.newProject.users.push(this.confirmedUser);
+        this.removeFromAvailableUsers(this.confirmedUser);
+      }
+    },
```

Comment:
Can you add some comments for alle these branches or even better add some descriptive functions for these checks.